scripts/cc-cli/src/globalSetup.ts

```diff
@@ -0,0 +1,134 @@
+): KeyringPair => {
+  switch (who) {
+    case "lender":
+      return keyring.addFromUri("//Alice");
+    case "borrower":
+      return keyring.addFromUri("//Bob");
+    case "sudo":
+      return keyring.addFromUri("//Alice");
+    default:
+      throw new Error(`Unexpected value "${who}"`); // eslint-disable-line
+  }
+};
+
+export async function deployCtcToken(existingAddress: string | undefined) {
+  const provider = new JsonRpcProvider(
+    (global as any).CREDITCOIN_ETHEREUM_NODE_URL,
+  );
+  const deployer = new Wallet(
+    (global as any).CREDITCOIN_CTC_DEPLOYER_PRIVATE_KEY,
+    provider,
+  );
+
+  const factory = new ContractFactory(
+    CtcArtifact.abi,
+    CtcArtifact.bytecode,
+    deployer,
+  );
+
+  let ctcToken: GluwaCreditVestingToken;
+
+  if (existingAddress !== undefined) {
+    ctcToken = factory.attach(existingAddress) as GluwaCreditVestingToken;
+  } else {
+    const deployerAddress = await deployer.getAddress();
+    ctcToken = (await factory.deploy(
+      deployerAddress,
+      deployerAddress,
+    )) as GluwaCreditVestingToken;
+  }
+
+  process.env.CREDITCOIN_CTC_CONTRACT_ADDRESS = ctcToken.address;
+
+  // burn 3 CTC
+  const tx = await ctcToken.burn((3 * CREDO_PER_CTC).toString());
+  const txHash = tx.hash;
+  await tx.wait();
+
+  console.log("Burn Tx Hash", txHash);
+  process.env.CREDITCOIN_CTC_BURN_TX_HASH = txHash;
+}
+
+export function setArg(key: string, value: any) {
+  (global as any)[key] = value;
+}
+
+export function setArgIfUndef(key: string, value: any) {
+  if ((global as any)[key] === undefined) {
+    setArg(key, value);
+  }
+}
+
+const globalDefaults = new Map<string, any>([
+  ["CREDITCOIN_EXECUTE_SETUP_AUTHORITY", true],
+  ["CREDITCOIN_CREATE_SIGNER", createSigner],
+  ["CREDITCOIN_ETHEREUM_NODE_URL", "http://127.0.0.1:8545"],
+  [
+    "CREDITCOIN_CTC_DEPLOYER_PRIVATE_KEY",
+    "0xabf82ff96b463e9d82b83cb9bb450fe87e6166d4db6d7021d0c71d7e960d5abe",
+  ],
+  [
+    "CREDITCOIN_API_URL",
+    `ws://127.0.0.1:${
+      process.env.CREDITCOIN_WS_PORT ? process.env.CREDITCOIN_WS_PORT : "9944"
+    }`,
+  ],
+]);
+
+export function arg(key: string) {
+  return (global as any)[key];
+}
+
+export async function setAuthorities() {
+  const api = await ApiPromise.create({
+    provider: new WsProvider((global as any).CREDITCOIN_API_URL),
+    throwOnConnect: true,
+  });
+  if ((global as any).CREDITCOIN_EXECUTE_SETUP_AUTHORITY) {
+    const keyring = new Keyring({ type: "sr25519" });
+    const sudo = (global as any).CREDITCOIN_CREATE_SIGNER(keyring, "sudo");
+    await setupAuthority(api, sudo);
+  }
+  await api.disconnect();
+}
+
+export function retry(retries: any, executor: any): any {
+  console.log(`${retries as string} retries left!`);
+
+  if (typeof retries !== "number") {
+    throw new TypeError("retries is not a number");
+  }
+
+  return new Promise(executor).catch((error) =>
+    retries > 0 ? retry(retries - 1, executor) : Promise.reject(error),
+  );
+}
```

Comment:
```suggestion
export async function retry<T>(
  retries: number,
  func: () => PromiseLike<T>
): Promise<T> {
  console.log(`${retries} retries left!`);

  try {
    return await func();
  } catch (error) {
    if (retries > 0) {
      return await retry(retries - 1, func);
    }
    throw error;
  }
}
```

IMO it's only necessary to drop down to using `new Promise` and explicit resolve/reject callbacks when you need more control over the promise, such as when you're converting a callback-based API to promises (which is what we do in creditcoin-js, since polkadot-js uses observables which are effectively callback-based). If the caller needs that, then they can do it themselves.

99% of the time though you don't need that, and you just want to use normal async/await notation.